OrbitVulkanLayer/VulkanLayerController.h

```diff
@@ -0,0 +1,454 @@
+    if (dispatch_table_.IsDebugMarkerExtensionSupported(command_buffer)) {
+      dispatch_table_.CmdDebugMarkerBeginEXT(command_buffer)(command_buffer, marker_info);
+    }
+    CHECK(marker_info != nullptr);
+    submission_tracker_.MarkDebugMarkerBegin(command_buffer, marker_info->pMarkerName,
+                                             {
+                                                 .red = marker_info->color[0],
+                                                 .green = marker_info->color[1],
+                                                 .blue = marker_info->color[2],
+                                                 .alpha = marker_info->color[3],
+                                             });
+  }
+
+  void OnCmdDebugMarkerEndEXT(VkCommandBuffer command_buffer) {
+    submission_tracker_.MarkDebugMarkerEnd(command_buffer);
+    if (dispatch_table_.IsDebugMarkerExtensionSupported(command_buffer)) {
+      dispatch_table_.CmdDebugMarkerEndEXT(command_buffer)(command_buffer);
+    }
+  }
+
+  // ----------------------------------------------------------------------------
+  // Layer enumeration functions
+  // ----------------------------------------------------------------------------
+
+  [[nodiscard]] VkResult OnEnumerateInstanceLayerProperties(uint32_t* property_count,
+                                                            VkLayerProperties* properties) {
+    // Vulkan spec dictates that we are only supposed to enumerate ourselve.
+    if (property_count != nullptr) {
+      *property_count = 1;
+    }
+    if (properties != nullptr) {
+      snprintf(properties->layerName, sizeof(properties->layerName), kLayerName);
+      snprintf(properties->description, sizeof(properties->description), kLayerDescription);
+      properties->implementationVersion = kLayerImplVersion;
+      properties->specVersion = kLayerSpecVersion;
+    }
+
+    return VK_SUCCESS;
+  }
+
+  [[nodiscard]] VkResult OnEnumerateInstanceExtensionProperties(
+      const char* layer_name, uint32_t* property_count, VkExtensionProperties* /*properties*/) {
+    // Inform the client that we have no extension properties if this layer
+    // specifically is being queried.
+    if (layer_name != nullptr && strcmp(layer_name, kLayerName) == 0) {
+      if (property_count != nullptr) {
+        *property_count = 0;
+      }
+      return VK_SUCCESS;
+    }
+
+    // Vulkan spec mandates returning this when this layer isn't being queried.
+    return VK_ERROR_LAYER_NOT_PRESENT;
+  }
+
+  [[nodiscard]] VkResult OnEnumerateDeviceExtensionProperties(VkPhysicalDevice physical_device,
+                                                              const char* layer_name,
+                                                              uint32_t* property_count,
+                                                              VkExtensionProperties* properties) {
+    // If our layer is queried exclusively, we just return our extensions
```

Comment:
"exclusively"? Does it mean that if `layer_name == nullptr` then all extensions are requested? In that case add a comment about it.

Reply:
Done.